openssl/src/x509/mod.rs

```diff
@@ -390,6 +390,69 @@ impl X509Generator {
         }
     }
 
+    /// Sets the certificate public-key, then self-sign and return it
```

Comment:
Hi Steven,

I refactored the code to prevent DRY-pattern.

I removed the ca_generate() function and added a set_ca(). I extended the
struct of X509Generate with ca: Option<(&'ctx X509<'ctx>, &'ctx PKey)>
I think it is cleaner from usage point of view, but I had to add the
lifetime specifier of X509 to the X509Generate struct and all function who
return the X509Generate object.
I asked a new pull request. Any comments are welcome.

Greetings,
DDT

2016-06-19 2:02 GMT+02:00 Steven Fackler <notifications@github.com>:

> In openssl/src/x509/mod.rs
> <https://github.com/sfackler/rust-openssl/pull/411#discussion_r67608303>:
>
> > @@ -390,6 +390,69 @@ impl X509Generator {
> >          }
> >      }
> >
> > +    /// Sets the certificate public-key, then self-sign and return it
>
> It seems like a good bit of the code between this and sign (and maybe
> others?) can be shared.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sfackler/rust-openssl/pull/411/files/338a02cb2e01401368260ceee3a0a3f2735d3f2a#r67608303>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABEe89Rnrz_qJ2L86NZhekpd62bYXQ23ks5qNIcbgaJpZM4I20vR>
> .
>